### PR TITLE
Replace mkdirp with builtin recursive flag

### PR DESCRIPTION
**Pull request by TrySound**
In node 10.12 mkdir got new recursive flag which provides the same
functionality as mkdirp.

See here https://nodejs.org/api/fs.html#fs_fs_mkdir_path_options_callback

Also node 10 got builtin promisified `fs` utilities.

See here https://nodejs.org/api/fs.html#fs_fs_promises_api

### CODE REVIEW

File: src/BundleAnalyzerPlugin.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
+const fs = reqiore('fs');
```
Comment (th0r) [inline src/BundleAnalyzerPlugin.js:1]:
```suggestion
const fs = require('fs');
```

File: src/viewer.js [hunk 2]
```diff
@@ -187,8 +186,8 @@ async function generateJSONReport(bundleStats, opts) {
 
   if (!chartData) return;
 
-  mkdir.sync(path.dirname(reportFilename));
-  fs.writeFileSync(reportFilename, JSON.stringify(chartData));
+  await fs.promises.mkdir(path.dirname(reportFilename), { recursive: true });
+  await fs.promises.writeFileSync(reportFilename, JSON.stringify(chartData));
```
Comment (th0r) [inline src/viewer.js:190]:
```suggestion
  await fs.promises.writeFile(reportFilename, JSON.stringify(chartData));
```

**Review by th0r — CHANGES_REQUESTED** [review]

File: package.json [hunk 3]
```diff
@@ -43,7 +43,6 @@
     "filesize": "^6.1.0",
     "gzip-size": "^5.1.1",
     "lodash": "^4.17.20",
-    "mkdirp": "^1.0.4",
```
Comment (th0r) [inline package.json:46]: Removing this line is not enough - `package-lock.json` should also be updated. Run `npm rm mkdirp` to properly remove dependency please.

**Review by th0r — CHANGES_REQUESTED** [review]

**Review by th0r — APPROVED** [review]

**Comment by th0r** [comment]
`licence/cla` check is stuck for some reason. @valscion do you know how to fix it?

**Comment by th0r** [comment]
Thanks!

**Comment by valscion** [comment]
> `licence/cla` check is stuck for some reason. @valscion do you know how to fix it?

Closing and reopening the pull request triggers a new `licence/cla` check in case the service was down for some reason in the past — that's how I've dealt with these situations before 😅

**Comment by TrySound** [comment]
I just do empty commits in such cases

**Comment by valscion** [comment]
Released in v4.2.0 :relaxed:. Thank you for your contributions!

**Pull request merged**